autoextract_spiders/sessions.py

```diff
@@ -0,0 +1,29 @@
+
+from crawlera_session import RequestSession as _Session
+
+
+def update_redirect_middleware(settings):
+    if settings.getbool('CRAWLERA_ENABLED'):
+        redirect_mware = 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware'
+        pos = settings.get('DOWNLOADER_MIDDLEWARES_BASE').pop(redirect_mware)
+        DW_MIDDLEWARES = settings.get('DOWNLOADER_MIDDLEWARES')
+        DW_MIDDLEWARES['crawlera_session.CrawleraSessionRedirectMiddleware'] = pos
+
+
+class RequestSession(_Session):
```

Comment:
The custom class is needed mainly to make the integration switchable, as the current implementation fails when some requests don't have expected headers/metadata. We could probably migrate the logic to crawlera-sessions repo, and drop it from here: it doesn't make a lot of sense to setup sessions if Crawlera is disabled.